src/sources/file.rs

```diff
@@ -467,13 +530,13 @@ pub fn file_source(
     };
 
     let event_metadata = EventMetadata {
-        host_key: config
-            .host_key
-            .clone()
-            .unwrap_or_else(|| log_schema().host_key().to_string()),
+        host_key: config.host_key.clone().map_or_else(
+            || parse_value_path(log_schema().host_key()).ok(),
+            |key| key.path,
+        ),
         hostname: crate::get_hostname().ok(),
-        file_key: config.file_key.clone(),
-        offset_key: config.offset_key.clone(),
+        file_key: config.file_key.clone().and_then(|f| f.path),
+        offset_key: config.offset_key.clone().and_then(|f| f.path),
```

Comment:
mega nit but for consistency, could use `k.path` or `key.path` for all these

Reply:
https://github.com/vectordotdev/vector/pull/15174/commits/93702138ccbbf8b028752b0779d9161d74673575